qiskit/synthesis/__init__.py

```diff
@@ -77,8 +79,12 @@
     QDrift,
 )
 
+from .permutation import (
+    synth_permutation_depth_lnn_kms,
+    synth_permutation_basic,
```

Comment:
Is `synth_permutation_basic` better than `synth_permutation_full`?
(in other methods we used "full")

Reply:
I am happy to rename this, but I wanted to stress that this is the original algorithm we had in Qiskit rather than some yet another algorithm that works for all-to-all connectivity. Well, I am slightly in favor of keeping names shorter, so when a synthesis algorithms assumes special connectivity (such as LNN or grid), it may be useful to somehow represent this in the method's name, and if it does not - then just leave this out from the name. On the other hand, I am not sure how to name algorithms that work for any restricted connectivity. In any case, the name does capture the most important aspects of the routine: it's a synthesis algorithm for permutations. What do you think?